libraries/lib-wave-track/AudioSegmentSampleView.cpp

```diff
@@ -0,0 +1,34 @@
+#include "AudioSegmentSampleView.h"
+
+size_t
+AudioSegmentSampleView::Copy(float* buffer, size_t bufferSize) const
+{
+   size_t toWrite =
+      std::min(mLength.as_long_long(), static_cast<long long>(bufferSize));
```

Comment:
What is the proof that there is no loss from this narrowing conversion?

Please use `limitSampleBufferSize` which contains some assertions.

`sampleCount` is a wide integral type used for positions in a sample stream or durations, because you can overflow 32 bits at 44.1 kHz with only about 13.5 hours, unlikely but not impossible.

`size_t` is sufficient for buffer sizes that fit in memory.

Reply:
Only the lesser of `bufferSize` and `GetSampleCount` is written, instead of filling the buffer up to `bufferSize` with zeroes as needed.

That behavior is not obvious to guess from the un-commented header file.  So please comment.